src/JsonRpc/JsonRpcServerServiceCollectionExtensions.cs

```diff
@@ -79,23 +81,51 @@ internal static IContainer AddJsonRpcMediatR(this IContainer container)
             container.RegisterMany(new[] { typeof(IMediator).GetAssembly() }, Registrator.Interfaces, Reuse.ScopedOrSingleton);
             container.RegisterMany<RequestContext>(Reuse.Scoped);
             container.RegisterDelegate<ServiceFactory>(context => context.Resolve, Reuse.ScopedOrSingleton);
+            container.Register(typeof(IRequestHandler<,>), typeof(RequestHandler<,>));
```

Comment:
This replacement code ensure that handlers can be added to the container as their MediatR interface `IRequestHandler<,>` and that will override the default registration, this is fine.
The decorator will ensure that even if a handler is registered that the handler from the current request will be used instead.